external_codes/catch/complex_approx.hpp

```diff
@@ -42,10 +41,7 @@ class ComplexApprox
     friend bool operator == (double const& lhs, ComplexApprox const& rhs)
     {
         bool is_equal = rhs.approx_compare(lhs, rhs.m_value.real());
-        if (!rhs.m_compare_real_only)
-        {
-          is_equal &= rhs.approx_compare(0.0, rhs.m_value.imag());
-        }
+        is_equal &= rhs.approx_compare(0.0, rhs.m_value.imag());
```

Comment:
The comparison between real lhs value and a complex rhs is invalid. ComplexApprox should always check both the real and the complex part. Please also remove the next function mixing complex and real.

Reply:
Fixed.